src/test/components/ProfileCallTreeView.test.js

```diff
@@ -97,15 +95,68 @@ describe('calltree/ProfileCallTreeView', function() {
         .join('\n')
     );
     const store = storeWithProfile(profile);
-    const calltree = renderer.create(
+    const calltree = mount(
       <Provider store={store}>
         <ProfileCallTreeView />
-      </Provider>,
-      { createNodeMock }
+      </Provider>
     );
 
     expect(calltree).toMatchSnapshot();
   });
+
+  it('reacts properly to up/down navigation keys', () => {
+    function setup() {
+      // This makes the bounding box large enough so that we don't trigger
+      // VirtualList's virtualization.
+      jest
+        .spyOn(HTMLElement.prototype, 'getBoundingClientRect')
+        .mockImplementation(() => getBoundingBox(1000, 2000));
+
+      // This generates a profile where function "name<i + 1>" is present
+      // <length - i> times, which means it will have a self time of <length - i>
+      // ms. This is a good way to control the order we'll get in the call tree
+      // view: function "name1" will be first, etc.
+      const profileString = Array.from({ length: 100 }).reduce(
+        (result, func, i, array) => {
+          const funcName = `name${i + 1}  `;
+          result += funcName.repeat(array.length - i);
+          return result;
+        },
+        ''
+      );
+
+      const { profile } = getProfileFromTextSamples(profileString);
+      const store = storeWithProfile(profile);
+      const calltree = mount(
+        <Provider store={store}>
+          <ProfileCallTreeView />
+        </Provider>
+      );
+
+      return {
+        simulateKey: (key: string) =>
+          calltree.find('div.treeViewBody').simulate('keydown', { key }),
+        selectedText: () =>
+          calltree.find('.treeViewRowScrolledColumns.selected').text(),
+      };
+    }
+
+    const { simulateKey, selectedText } = setup();
+
+    expect(selectedText()).toBe('name1');
+    simulateKey('ArrowDown');
+    expect(selectedText()).toBe('name2');
+    simulateKey('PageDown');
+    expect(selectedText()).toBe('name17'); // 15 rows below
+    simulateKey('End');
+    expect(selectedText()).toBe('name100');
+    simulateKey('ArrowUp');
+    expect(selectedText()).toBe('name99');
+    simulateKey('PageUp');
+    expect(selectedText()).toBe('name84'); // 15 rows above
+    simulateKey('Home');
+    expect(selectedText()).toBe('name1');
```

Comment:
Note I didn't test `'ArrowLeft'` and `'ArrowRight'` because it would need a separate profile and a separate test, and this wasn't the goal of this code change. But if you want I can add one.